### DISCUSSION TITLE
How to get a prediction from a new session not included in dataset? (Session-based recommendation)

### DISCUSSION THREAD

**Discussion by elloza** [RUCAIBox/RecBole#1305] [Q&A]
Hello,

I would like to use RecBole's already trained models (sequential) to predict the next item from some interactions but from a session_id that is not in the dataset. (Session-based recommendation)

Is it possible to do this with this library?

This question is related to [this](https://github.com/RUCAIBox/RecBole/issues/632#issuecomment-757537173) and [this](https://github.com/RUCAIBox/RecBole/issues/506#issuecomment-752662864).

"Given sequence of items, calculate the scores between the given sequence and all candidate items."

The main question is how to use RecBole with sequential models to get a prediction of a next item given a list of previous items and without a session_id (or user_id)?.

Thank you so much!

**Comment by Sherry-XLL** [comment]
Hello, @elloza

In the implementation of RecBole's sequential models, session_id (or user_id) is not needed for predicting. 

If you want to calculate the scores between the given sequence and all candidate items, you can just process the sequence of items into an interaction with `ITEM_SEQ` and `ITEM_SEQ_LEN`. As is calculated in `full_sort_predict`, `scores` are the scores between the given sequence and all candidate items.

https://github.com/RUCAIBox/RecBole/blob/ddff6211d030906f502a073a9f2b9e2250e93664/recbole/model/sequential_recommender/gru4rec.py#L113-L119

Best Wishes.
